### ISSUE TITLE
Vert.x grpc - newVertxStub vs newStub

### ISSUE THREAD

**Issue by saranshvarshneya** [vert-x3/vertx-grpc#28]
I am a beginner with grpc and had the following basic doubt as to is there any restriction that we need to use "newVertxStub" on the client side when we use Vertx grpc and not "newStub" and similarly on the server side to use VertxImpl rather than Impl class for the service . I am just confused since the method name mentions Vertx so is there a need/benefit that only the ones with Vertx in the name are the correct methods to be used.
Also if there is no such need then what is the difference between the two and which one is supposed to be used in what scenarios? 

I needed this answer urgently and had put up a question on google groups but did not get any response so had to create a issue here . 
https://groups.google.com/forum/#!topic/vertx/hj6IyfeZLm4

**Comment by andrm** [comment]
The difference between the two is that the vertxStubs use the vert.x infrastructure to execute the handlers - things that you want to happen when a request or response from or to a grpc service comes in.

I have a project where I use both: On the client side, I work with "newStub" and have blocking actions, like "response = myGrpcServiceStub.getInfo(arg1)". This is the normal, sequential, synchronous way of doing RPC. For my client, this is fine.

On the server I do want to use the Vert.x goodness, like using the eventbus, getting asynchronous answers from other http services etc. Here I used "newVertxStub()". The advantage is that I can be fully asynchronous, no threads being held up by waiting for reading a file or a http response. This scales very well.

The first paragraphs of the Vertx documentation als apply to the GRPC implementation also.

**Comment by saranshvarshneya** [comment]
But internally both(newStub and newVertxStub) of them use "asyncBidiStreamingCall()" , do you mean that a new task shall not be created and executed on an event loop in case of newStub?

**Comment by andrm** [comment]
I think the GRPC code that is used to talk to the other side is the same (Netty eventloop), the local case is different: Vertx uses its eventloop to deal with a response, no synchronous wait.

Please try to use both APIs and use the one that fits your needs better. If you don't need vertx, use the normal grpc-java code.
